src/utils/arrayUtils.ts

```diff
@@ -41,3 +41,35 @@ export function removeFalsy<const T>(
 ): T[] {
   return array.filter((e) => e) as T[]
 }
+
+export type RecordKeyType = string | number | symbol
+export function arrayToRecord<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(
+  array: T[],
+  selector: K,
+  arrayValue?: true,
+): Record<T[K], T[]>
+export function arrayToRecord<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(
+  array: T[],
+  selector: K,
+  arrayValue: false,
+): Record<T[K], T>
+export function arrayToRecord<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(
```

Comment:
Lodash calls this "keyBy", I think it's more intuitive name. See https://www.geeksforgeeks.org/lodash-_-keyby-method/

alternative suggestion: keyArrayBy

Reply:
Wow, you are really quick! thanks Igor 🙇 

Yeah, your options are better 🙌, which one do you think is more intuitive `keyBy` or `keyArrayBy`? I am more for `keyArrayBy` but I am not sure 🤔

Reply:
yeah, I like keyArrayBy too